src/redist/targets/GenerateMSIs.targets

```diff
@@ -13,31 +13,34 @@
     <PropertyGroup>
       <CombinedFrameworkSdkHostBundleEngineName>$(ArtifactsTmpDir)/$(ArtifactNameWithVersionCombinedHostHostFxrFrameworkSdk)-engine.exe</CombinedFrameworkSdkHostBundleEngineName>
 
-      <SdkGenerateMsiPowershellScript>$(MSBuildThisFileDirectory)packaging/windows/clisdk/generatemsi.ps1</SdkGenerateMsiPowershellScript>
-      <SdkPlaceholderGenerateMsiPowershellScript>$(MSBuildThisFileDirectory)packaging/windows/clisdk/generatesdkplaceholdermsi.ps1</SdkPlaceholderGenerateMsiPowershellScript>
-      <SdkStableFileIdForApphostTransform>$(MSBuildThisFileDirectory)packaging/windows/clisdk/stablefileidforapphosttransform.xslt</SdkStableFileIdForApphostTransform>
-      <SdkGenerateBundlePowershellScript>$(MSBuildThisFileDirectory)packaging/windows/clisdk/generatebundle.ps1</SdkGenerateBundlePowershellScript>
-      <GenerateNupkgPowershellScript>$(MSBuildThisFileDirectory)packaging/windows/clisdk/generatenupkg.ps1</GenerateNupkgPowershellScript>
+      <SdkPkgSourcesRootDirectory>$(MSBuildThisFileDirectory)../packaging/windows</SdkPkgSourcesRootDirectory>
+      <SdkGenerateMsiPowershellScript>$(SdkPkgSourcesRootDirectory)/generatemsi.ps1</SdkGenerateMsiPowershellScript>
+      <SdkPlaceholderGenerateMsiPowershellScript>$(SdkPkgSourcesRootDirectory)/generatesdkplaceholdermsi.ps1</SdkPlaceholderGenerateMsiPowershellScript>
+      <SdkStableFileIdForApphostTransform>$(SdkPkgSourcesRootDirectory)/stablefileidforapphosttransform.xslt</SdkStableFileIdForApphostTransform>
+      <SdkGenerateBundlePowershellScript>$(SdkPkgSourcesRootDirectory)/generatebundle.ps1</SdkGenerateBundlePowershellScript>
+      <GenerateNupkgPowershellScript>$(SdkPkgSourcesRootDirectory)/generatenupkg.ps1</GenerateNupkgPowershellScript>
 
-      <FinalizerExe>$(ArtifactsDir)bin/finalizer/$(Architecture)/$(Configuration)/bin/finalizer.exe</FinalizerExe>
+      <FinalizerExe>$(ArtifactsDir)bin/finalizer/$(Configuration)/bin/finalizer.exe</FinalizerExe>
```

Comment:
The addition of `$(Architecture)` is no longer part of the path as part of the finalizer build changes. But it wasn't necessary anyway was multiple architectures are not built on the same machine.

Reply:
this might be true in CI, but I'm not so sure about locally. maybe just keep this?

Reply:
But what scenario would there be to build finalizer locally for multiple architectures as it is only used for MSI creation? Is that a use case of making multi-architecture unsigned MSIs?

Reply:
I was more thinking about e.g. doing an x86 build and then doing an x64 build later on, presumably the binary would be overwritten? Nevermind if you don't think it's an issue.

Reply:
The finalizer is nothing to do with MSI creation. It's an extra executable that is linked into the standalone EXE bundles and because it accesses things like the registry, the architecture is critical to decide upon default hives, etc. It runs when the standalone SDK is removed to clean up workloads that were installed by the CLI.

Manually verifying changes to the finalizer would be one case where you would one to build x86 and x64 locally and then test on a VM. Normally in CI and non-CI builds we'll have separate legs, but this could be confusing if you're doing local builds to test

Reply:
![image](https://media2.giphy.com/media/v1.Y2lkPTc5MGI3NjExbXl2MHZxNTZkazZmMGIwaWRmZXZhcjI4c2hlZXgyNDBvZWZwcWdwNiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/a5viI92PAF89q/giphy.gif)

Took me some time staring at what you said and rifling through things. I realized I've never known what the actual difference between the .EXE installer and the .MSI installer. Like, from a software user POV. In my monkey brain, I always assumed that .MSI is the "blessed" Microsoft installation platform. Prior to .MSI, you'd use different installer platforms (one that comes to mind is the Nullsoft one, popularized by WinAmp) and those made .EXEs. I've also used InstallShield at a previous job and helped configure the installer for their software.

Here, we produce both an .EXE installer and .MSI installer. My assumption was that the .EXE installer basically just "wraps" the .MSI installer. When I say "MSI creation", my brain equates that to "installer creation" because I didn't really see a difference.

Also, in my defense, this is in the file `GenerateMSIs.targets`, so I wonder why I thought it was related to the creation of the MSI... 🤔 The world may never know.

*Also also* in my defense, I didn't **explicitly** change the output path logic. Meaning, I didn't *remove* `$(Architecture)` from the output path as a conscious decision. I did search the repo to try to figure out why it changed, but I didn't find anything that explained it. The only way would be to look though the current .binlogs and see how the path is constructed. It wasn't a priority as these changes were originally done in the SDK repo PR, not here. I only moved these changes here later. Seeing that the path still has this change, though, I can do some digging and try to figure out what happened.

Reply:
It only becomes risky if we ever merge the different legs for architectures. Pulling in the x86 finalizer into the arm64 installer could have really bad side effects on users machines. Similarly, pulling in the x64 finalizer into the x86 bundle would lead to a failed install because it can't run on a 32-bit OS. 

Until the introduction of optional workloads, we only chained MSIs in the bundle. The finalizer was the first time we needed a non-MSI based package to be chained in so the MSI targets was a bit of convenient cheat to keep all the bundle packages together regardless of the type of package.

Reply:
So, spending an hour digging between the binlogs for the finalizer projects...

As an example for the `arm64` build:

| Property | Current Builds | This PR
| --- | --- | ---
| `_CMakePassArchitectureToGenerator` | false | false
| `_UsingDefaultPlatformTarget` | *[unset]* | true
| `Architecture` | arm64 | arm64
| `BuildArchitecture` | x64 | x64
| `ConsiderPlatformAsProcessorArchitecture` | true | true
| `Platform` | arm64 | arm64
| `PlatformName` | arm64 | AnyCPU
| `Platforms` | AnyCPU | AnyCPU
| `PlatformTarget` | arm64 | *[unset]*
| `ProcessorArchitecture` | arm64 | **AMD64**
| `ProcessorArchitectureAsPlatform` | *[unset]* | arm64
| `TargetedSDKArchitecture` | arm64 | **AMD64**

I have no earthly idea how any of this information is used. I'm also still not closer to understanding how this information affects the `OutputPath`.

Reply:
Okay, it might be because I moved this logic from the outer project (.csproj) to the inner one (.nativeproj):

https://github.com/dotnet/installer/blob/68bc63885822bf58198adfb1394afe68bdf4a5cc/eng/native.proj#L4

I'll try moving it to the outer project and see what happens.

Reply:
More testing and investigation (again, there's a reason I didn't do this originally because it is time consuming), it seems that Arcade is modifying the output path. So, it happens like this:
- You need to set `Platform` either at `Directory.Build.props` time or earlier
- If you do, the Arcade SDK runs `RepoDefaults.props` that sets `PlatformName`: https://github.com/dotnet/arcade/blob/d4a7b37f268cb5fe89f4472ef5341629cd040dfe/src/Microsoft.DotNet.Arcade.Sdk/tools/RepoDefaults.props#L10-L11
- Then, the Arcade SDK runs `ProjectLayout.props` and if `PlatformName` is set, it modifies the `OutputPath` to include a `PlatformName` directory: https://github.com/dotnet/arcade/blob/d4a7b37f268cb5fe89f4472ef5341629cd040dfe/src/Microsoft.DotNet.Arcade.Sdk/tools/ProjectLayout.props#L15-L16

So, I have an idea to fix this. Let me try something.

Reply:
I figured out my fundamental confusion. My assumption had been that this content in the Directory.Build.props sets Architecture:
https://github.com/dotnet/installer/blob/fc9e980c5fb8cc8767c2926934d823ac0d0bfa69/Directory.Build.props#L10-L17

### IT DOES NOT.

Just a few lines above that, the Arcade SDK is loaded:
https://github.com/dotnet/installer/blob/fc9e980c5fb8cc8767c2926934d823ac0d0bfa69/Directory.Build.props#L3

But Architecture is needed to be set prior to calling the Arcade SDK. So where is it set? It is set here (for local builds):
https://github.com/dotnet/installer/blob/fc9e980c5fb8cc8767c2926934d823ac0d0bfa69/run-build.ps1#L18

`build.cmd` calls `run-build.ps1` which calls `eng\common\build.ps1`. Local builds go down this path.

CI builds, however, set `buildArchitecture` for for the `eng/build.yml` template. For example:
https://github.com/dotnet/installer/blob/fc9e980c5fb8cc8767c2926934d823ac0d0bfa69/.vsts-ci.yml#L177

Then, `buildArchitecture` is set to Architecture:
https://github.com/dotnet/installer/blob/fc9e980c5fb8cc8767c2926934d823ac0d0bfa69/eng/build.yml#L189

### THEREFORE

Currently, if you wanted to build finalizer without using `build.cmd`, it won't output properly. I'll fix that now.

Reply:
I've now fixed this. I've added a `Directory.Build.props` into the finalizer folder and see the details for how Platform is set in the root's Directory.Build.props.